src/sinks/util/batch.rs

```diff
@@ -116,6 +137,19 @@ impl BatchSettings {
             ..self
         }
     }
+
+    // Would like to use `trait From` here, but that results in
```

Comment:
So if you can limit `B2` such that Rust can know `BatchSettings` is not a valid input you might be able to do this?

Reply:
AFAICT the issue is excluding `B1` from being the same as `B2`, because where they are the same this collapses to just `impl From<T> for T` (where T is `BatchSettings<B>`) which is pre-defined as an identity transform.

Reply:
Aw shucks, ok. :(